homeassistant/components/gios/sensor.py

```diff
@@ -230,11 +230,11 @@ def native_value(self) -> StateType:
     @property
     def available(self) -> bool:
         """Return if entity is available."""
-        available = super().available
         sensor_data = getattr(self.coordinator.data, self.entity_description.key)
+        available = super().available and bool(sensor_data)
 
         # Sometimes the API returns sensor data without indexes
-        if self.entity_description.subkey:
+        if self.entity_description.subkey and available:
             return available and bool(sensor_data.index)
```

Comment:
An index can also be 0 ?

Reply:
`index` is a non-empty string (`good`, `very_good`, `moderate`, etc.) or `None` https://github.com/bieniu/gios/blob/d836f0c048bf2eb2847559467c5b05f264c42641/gios/model.py#L12